### PR TITLE
Fix rxjs flow issues

### PR DESCRIPTION
**Pull request by lgeiger**
Thanks to @jayphelps we were able to fix almost all of the flow issues left in #3399 🎉

### CODE REVIEW

File: applications/desktop/src/notebook/epics/loading.js [hunk 1]
```diff
@@ -145,7 +152,7 @@ export const launchKernelWhenNotebookSetEpic = (
 ) =>
   action$.pipe(
     ofType(actionTypes.FETCH_CONTENT_FULFILLED),
-    map((action: actionTypes.FetchContentFulfilled) => {
+    mergeMap((action: actionTypes.FetchContentFulfilled) => {
```
Comment (lgeiger) [inline applications/desktop/src/notebook/epics/loading.js:155]: Sadly writing this with `map->filter->map` made flow fail again. Merge map works fine though.

File: packages/core/src/epics/contents.js [hunk 2]
```diff
@@ -14,10 +14,6 @@ import * as actionTypes from "../actionTypes";
 import * as selectors from "../selectors";
 import type { ContentRef, AppState } from "../state";
 
-// Flow complains otherwise since of would return an rxjs$Observable rather than an ActionsObservable
-// Ask Jay Phelps about this ;)
-const of = ActionsObservable.of;
-
 export function fetchContentEpic(
   action$: ActionsObservable<redux$Action>,
```
Comment (lgeiger) [inline packages/core/src/epics/contents.js:18]: We actually could switch to `Observable<redux$Action>` everywhere.

File: packages/fs-observable/src/index.js [hunk 3]
```diff
@@ -23,32 +23,16 @@ export const unlinkObservable = (path: string) =>
...
+export const createSymlinkObservable = (target: string, path: string) =>
   unlinkObservable(path).pipe(
     mergeMap(() => createNewSymlinkObservable(target, path))
   );
 
-// NOTE: Flow types here match our current usage rather than inferring it,
-// due to bindNodeCallback not being typed
-export const readFileObservable: (
-  pth: string
-) => Observable<Buffer> = bindNodeCallback(fs.readFile);
```
Comment (lgeiger) [inline packages/fs-observable/src/index.js:44]: `bindNodeCallback` is now properly typed.

**Review by lgeiger — COMMENTED** [review]
Flow error messages are really confusing. I'm happy we got this fixed now.

File: packages/fs-observable/src/index.js (hunk 3 above)
Comment (rgbkrk) [reply packages/fs-observable/src/index.js:44]: Wahoo!

**Review by rgbkrk — APPROVED** [review]
Mega thank you to both of you!

**Pull request merged**